rulette-postgres-provider/src/main/java/com/github/kislayverma/rulette/postgres/PostgresDataProvider.java

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright 2016 kislay.verma.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.github.kislayverma.rulette.postgres;
+
+import com.github.kislayverma.rulette.core.data.IDataProvider;
+import com.github.kislayverma.rulette.core.metadata.RuleInputMetaData;
+import com.github.kislayverma.rulette.core.metadata.RuleSystemMetaData;
+import com.github.kislayverma.rulette.core.rule.Rule;
+import com.github.kislayverma.rulette.core.ruleinput.type.RuleInputType;
+import com.github.kislayverma.rulette.postgres.dao.DataSource;
+
+import java.io.IOException;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ *
+ * @author kislay.verma
+ */
+public class PostgresDataProvider implements IDataProvider {
+    private final Map<String, RuleSystemMetaData> metaDataMap;
+
+    public PostgresDataProvider(String datasourceUrl) throws IOException, SQLException {
+        metaDataMap = new ConcurrentHashMap<>();
+        DataSource.init(datasourceUrl);
+    }
+
+    private Connection getConnection() throws SQLException, IOException {
+        return DataSource.getInstance(null).getConnection();
+    }
+
+    @Override
+    public List<Rule> getAllRules(String ruleSystemName) throws SQLException, Exception {
+        List<Rule> rules = new ArrayList<>();
+
+        Statement statement = getConnection().createStatement();
+        ResultSet resultSet =
+            statement.executeQuery("SELECT * " + " FROM " + getRuleSystemMetaData(ruleSystemName).getTableName());
+
+        if (resultSet != null) {
+            rules = convertToRules(resultSet, getRuleSystemMetaData(ruleSystemName));
+        }
+
+        return rules;
+    }
+
+    @Override
+    public Rule saveRule(String ruleSystemName, Rule rule) throws SQLException, Exception {
```

Comment:
@kislayverma Take a look at https://spring.io/projects/spring-data I think that can save a lot of pain for this use case and easily gives integration with so many databases.

Reply:
Thanks for the suggestion @buchireddy. One of the ideas for Rulette is for it to be very, VERY lightweight hence I don't want to add any more frameworks than I absolutely must - and Spring was never known to be lightweight :)